swagger.yaml

```diff
@@ -3925,142 +7803,215 @@ definitions:
         type: string
         description: A CIDR that all hosts belonging to the cluster should have an interfaces with IP address that belongs to this CIDR. The api_vip belongs to this CIDR.
         pattern: '^(?:(?:(?:[0-9]{1,3}\.){3}[0-9]{1,3}\/(?:(?:[0-9])|(?:[1-2][0-9])|(?:3[0-2])))|(?:(?:[0-9a-fA-F]*:[0-9a-fA-F]*){2,})/(?:(?:[0-9])|(?:[1-9][0-9])|(?:1[0-1][0-9])|(?:12[0-8])))$'
-        x-nullable: true
-      pull_secret:
+      ingress_vip:
         type: string
-        description: The pull secret obtained from Red Hat OpenShift Cluster Manager at cloud.redhat.com/openshift/install/pull-secret.
-        x-nullable: true
+        pattern: '^(?:(?:(?:[0-9]{1,3}\.){3}[0-9]{1,3})|(?:(?:[0-9a-fA-F]*:[0-9a-fA-F]*){2,}))$'
+        description: The virtual IP used for cluster ingress traffic.
       ssh_public_key:
         type: string
         description: SSH public key for debugging OpenShift nodes.
-        x-nullable: true
+      http_proxy:
+        type: string
+        description: |
+          A proxy URL to use for creating HTTP connections outside the cluster.
+          http://\<username\>:\<pswd\>@\<ip\>:\<port\>
+      https_proxy:
+        type: string
+        description: |
+          A proxy URL to use for creating HTTPS connections outside the cluster.
+          http://\<username\>:\<pswd\>@\<ip\>:\<port\>
+        x-go-custom-tag: gorm:"column:https_proxy"
+      no_proxy:
+        type: string
+        description: A comma-separated list of destination domain names, domains, IP addresses, or other network CIDRs to exclude from proxying.
+      status:
+        type: string
+        description: Status of the OpenShift cluster.
+        enum:
+          - insufficient
+          - ready
+          - error
+          - preparing-for-installation
+          - pending-for-input
+          - installing
+          - finalizing
+          - installed
+          - adding-hosts
+          - cancelled
+          - installing-pending-user-action
+      status_info:
+        type: string
+        x-go-custom-tag: gorm:"type:varchar(2048)"
+        description: Additional information pertaining to the status of the OpenShift cluster.
+      status_updated_at:
+        type: string
+        format: date-time
+        x-go-custom-tag: gorm:"type:timestamp with time zone"
+        description: The last time that the cluster status was updated.
+      hosts:
+        x-go-custom-tag: gorm:"foreignkey:ClusterID;association_foreignkey:ID"
```

Comment:
since we are going to use agents table , i thing that this field can be defined with gorm:"-"
we are going to fill those fields only when we return the reply, by reading the agents field